### PR TITLE
ENH: Environment Analysis - Animate Wind Gust Distribution

### PR DESCRIPTION
**Pull request by MateusStano**
## Pull request type

Please check the type of change your PR introduces:

- [x] Code base additions (bugfix, features)
- [ ] Code maintenance (refactoring, formatting, renaming, tests)
- [ ] ReadMe, Docs and GitHub maintenance
- [ ] Other (please describe):

## Pull request checklist

Please check if your PR fulfills the following requirements, depending on the type of PR:

- Code base additions (for bug fixes / features):

  - [ ] Tests for the changes have been added
  - [ ] Docs have been reviewed and added / updated if needed
  - [x] Lint (`black rocketpy`) has passed locally and any fixes were made
  - [x] All tests (`pytest --runslow`) have passed locally

## What is the new behavior?

Method `EnvironmentAnalysis.animate_wind_gust_distribution_over_average_day` is now implemented.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by giovaniceotto** [comment]
@PatrickSampaioUSP, Stano and I performed a first implementation and it is ready for review. Here is a preview of the outcome I got:

wind_gust_distribution_animation

**Comment by PatrickSampaioUSP** [comment]
When executing `animate_wind_gust_distribution_over_average_day` i'm receiving the following error:

`AttributeError: 'list' object has no attribute 'set_animated'`

File: rocketpy/EnvironmentAnalysis.py [hunk 1]
```diff
@@ -610,11 +612,148 @@ def calculate_average_day_wind_rose(self):
...
+    def plot_wind_gust_distribution_over_average_day(self):
+        """Plots shown in the animation of how the wind gust distribution varies throughout the day."""
+        ...
+        # Gather animation data
+        average_wind_gust_at_given_hour = {}
+        for hour in list(self.surfaceDataDict.values())[0].keys():
+            wind_gust_values_for_this_hour = []
+            for dayDict in self.surfaceDataDict.values():
+                try:
+                    wind_gust_values_for_this_hour += [dayDict[hour]["surfaceWindGust"]]
```
Comment (PatrickSampaioUSP) [inline rocketpy/EnvironmentAnalysis.py:624]: if you do `dayDict.get(hour, {}).get("surfaceWindGust", 0)` you can take out this try/except which I think improves the readability of the function

Comment (PatrickSampaioUSP) [inline rocketpy/EnvironmentAnalysis.py:624]: You can substitute this try/except block by:

`[dayDict.get('hour', {}).get('surfaceWindGust', None)] if dayDict.get('hour', {}).get('surfaceWindGust', None) else []`

But honestly it is not the most readable sentence. So if you want to maintain the try/excepet block ok, but there is the option to change for this sentence, up to you

**Comment by PatrickSampaioUSP** [comment]
@MateusStano my review is ready, waiting for your response

**Comment by giovaniceotto** [comment]
> When executing `animate_wind_gust_distribution_over_average_day` i'm receiving the following error:
> 
> `AttributeError: 'list' object has no attribute 'set_animated'`

I am not able to reproduce this error here. Would you mind sending the entire traceback showing which line is triggering this error?

Since I can't reproduce it here, I would guess it has to do with different versions of matplotlib. I am currently running the latest version. I will try older ones to check. If you could also detail which version you are using, that would help.

**Comment by giovaniceotto** [comment]
Going back to Matplotlib version 3.2, which is from March 4, 2020: `AttributeError: 'NoneType' object has no attribute '_send_event'`

**Comment by giovaniceotto** [comment]
@PatrickSampaioUSP, can you confirm that with and updated version of matplotlib you were able to run the code?

**Comment by PatrickSampaioUSP** [comment]
Sorry for the delay, I tried to update the matplotlib version but I'm still receiving an error

**Comment by PatrickSampaioUSP** [comment]
Sorry for the delay, I tried to update the matplotlib version but I'm still receiving an error

Screenshot from 2022-05-16 23-49-09

**Comment by PatrickSampaioUSP** [comment]
The stack trace error

[stack.txt](https://github.com/Projeto-Jupiter/RocketPy/files/8703660/stack.txt)

**Comment by giovaniceotto** [comment]
Pretty sure I have fixed the error now! After I was inspired by @MateusStano, I was able to reproduce the exact same `AttributeError: 'list' object has no attribute 'set_animated'` on my PC by running a script instead of a notebook.

It turns out I was missing a `*` operator in two lines of code... Now that I see this, I don't even know how was it working before on notebooks! Anyways...

@PatrickSampaioUSP and @MateusStano, could you give this PR one more chance, run it and check if there is any problem?

Here is a sample code to make things easier:

```python
from rocketpy.EnvironmentAnalysis import EnvironmentAnalysis
from datetime import datetime

test_analysis = EnvironmentAnalysis(
    start_date=datetime(2010, 6, 15),
    end_date=datetime(2010, 6, 30),
    latitude=33,
    longitude=-107,
    elevation=1400,
    pressureLevelDataFile="data\weather\spaceport_america_pressure_levels_reanalysis_2002_2021.nc",
    surfaceDataFile="data\weather\spaceport_america_single_level_reanalysis_2002_2021.nc",
)

test_analysis.animate_wind_gust_distribution_over_average_day()
```

**Review by PatrickSampaioUSP — APPROVED** [review]

**Pull request merged**